media-plugins/gkrellmpc/gkrellmpc-0.1_beta10-r1.ebuild

```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit gkrellm-plugin toolchain-funcs
+
+DESCRIPTION="A gkrellm plugin to control the MPD (Music Player Daemon)"
+HOMEPAGE="http://mpd.wikia.com/wiki/Client:GKrellMPC"
+SRC_URI="http://mina.naguib.ca/dist/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="threads"
+
+RDEPEND="
+	app-admin/gkrellm:2[X]
+	net-misc/curl"
+DEPEND="${RDEPEND}"
+
+# Will open gkrellm in X11 display
+RESTRICT="test"
+
+PATCHES=( "${FILESDIR}/${PF}"-make-test.patch )
```

Comment:
The patch only really has an effect for "make test", but then there's `RESTRICT="test"` right above. I converted the `sed` line from `-r0` to a patch here.

Reply:
Can we unrestrict tests? How do they fail?

Reply:
Just commenting out the RESTRICT line, produces:

```
$ FEATURES=test ebuild ./gkrellmpc-0.1_beta10-r1.ebuild clean install
[snip]
>>> Test phase: media-plugins/gkrellmpc-0.1_beta10-r1
make -j5   test
gkrellm -p gkrellmpc.so

(gkrellm:7901): Gtk-WARNING **: 09:18:39.522: cannot open display:
make: *** [Makefile:93: test] Error 1
 * ERROR: media-plugins/gkrellmpc-0.1_beta10-r1::gentoo failed (test phase):
 *   Make test failed. See above for details.
```

Reply:
I suppose the ebuild process is not permitted access to interact with X11.

Running the test command manually reveals the following problem in the plugin Install Log:

```
*** Command line plugin:
./gkrellmpc.so
	Error: ./gkrellmpc.so: undefined symbol: curl_easy_getinfo
```

Reply:
Can you try using the virtualx eclass?

Reply:
Sure.

Reply:
Sorry for being slow on this one. I changed the ebuild:

```diff
diff --git a/media-plugins/gkrellmpc/gkrellmpc-0.1_beta10-r1.ebuild b/media-plugins/gkrellmpc/gkrellmpc-0.1_beta10-r1.ebuild
index 22adb571df5..37616f9222d 100644
--- a/media-plugins/gkrellmpc/gkrellmpc-0.1_beta10-r1.ebuild
+++ b/media-plugins/gkrellmpc/gkrellmpc-0.1_beta10-r1.ebuild
@@ -3,7 +3,7 @@
 
 EAPI=8
 
-inherit gkrellm-plugin toolchain-funcs
+inherit gkrellm-plugin toolchain-funcs virtualx
 
 DESCRIPTION="A gkrellm plugin to control the MPD (Music Player Daemon)"
 HOMEPAGE="http://mpd.wikia.com/wiki/Client:GKrellMPC"
@@ -20,7 +20,7 @@ RDEPEND="
 DEPEND="${RDEPEND}"
 
 # Will open gkrellm in X11 display
-RESTRICT="test"
+#RESTRICT="test"
 
 PATCHES=( "${FILESDIR}/${PF}"-make-test.patch )
 
@@ -35,6 +35,10 @@ src_configure() {
 	tc-export CC
 }
 
+src_test() {
+	virtx make test
+}
+
 pkg_postinst() {
 	if use threads; then
 		elog "If you can't connect MPD, please unset USE threads."

```
Result:
```
>>> Test phase: media-plugins/gkrellmpc-0.1_beta10-r1
 * ERROR: media-plugins/gkrellmpc-0.1_beta10-r1::gentoo failed (test phase):
 *   (no error message)
 *
 * Call stack:
 *                        ebuild.sh, line 127:  Called src_test
 *   gkrellmpc-0.1_beta10-r1.ebuild, line  39:  Called virtx 'make' 'test'
 *                      environment, line 1982:  Called die
 * The specific snippet of code:
 *       XVFB=$(type -p Xvfb) || die;
 *
```
(I guess I don't have USE=xvfb on x11-base/xorg-server)